src/main/java/org/nastya/controller/UserController.java

```diff
@@ -16,23 +16,23 @@ public UserController(UserService userService) {
         this.userService = userService;
     }
 
-    @GetMapping("/{username}")
-    public ResponseEntity<User> getUserByUsername(@PathVariable String username) {
-        Optional<User> user = userService.findByUsername(username);
-        return user.map(ResponseEntity::ok)
+    @GetMapping("/username")
+    public ResponseEntity<UserDTO> getUserByUsername(@PathVariable String username) {
+        Optional<UserDTO> userDTO = userService.findByUsername(username);
+        return userDTO.map(ResponseEntity::ok)
                 .orElseGet(() -> ResponseEntity.notFound().build());
     }
 
     @PostMapping
-    public ResponseEntity<User> createUser(@RequestBody User user) {
-        if (userService.existsByUsername(user.getUsername())) {
+    public ResponseEntity<UserDTO> createUser(@RequestBody UserDTO userDTO) {
```

Comment:
The controller's method createUser has business logic.

For example:
if user exists do this (return bad request) if not do that (save)

Business has to be on the service level. Move it to UserService.